lib/open_food_network/reports/line_items.rb

```diff
@@ -51,14 +51,16 @@ def search_orders
         @order_permissions.visible_orders.complete.not_state(:canceled).search(@params[:q])
       end
 
-      def line_items_with_hidden_details(line_items)
-        editable_line_items = @order_permissions.editable_line_items.pluck(:id)
+      # From the line_items given, returns the ones that are editable by the user
+      def editable_line_items(line_items)
+        editable_line_items_ids = @order_permissions.editable_line_items.select(:id)
 
-        if editable_line_items.empty?
-          line_items
-        else
-          line_items.where('"spree_line_items"."id" NOT IN (?)', editable_line_items)
-        end
+        # Although merge could take a relation, here we convert line_items to array
+        #   because, if we pass a relation, merge will overwrite the conditions on the same field
+        #   In this case: the IN clause on spree_line_items.order_id from line_items
+        #     overwrites the IN clause on spree_line_items.order_id on editable_line_items_ids
+        # We convert to array the relation with less elements: line_items
+        editable_line_items_ids.merge(line_items.to_a)
```

Comment:
I came across this recently when fixing the Lettuceshare report. I solved it by using SQL:
https://github.com/openfoodfoundation/openfoodnetwork/pull/4168/files#diff-db07cecd37d0749b2162463301567e23L68-R72

I think that conditions in hashes get overridden and it's a bug. If you add the condition as a string as in `where("id in (?)")` it does not override the condition. In this case I'm wondering if you could do:
```rb
editable_line_items_ids.where("spree_line_items.id IN (?)", line_items.select(:id))
```

But looking at the code again, `line_items` is loaded into an array anyway. As long as the list is reasonably short, it's better to re-use that in-memory list instead of asking the database to compile the ids again.

Reply:
Yeah, thanks for sharing. It's really good to know this.
I will leave it as is. I think this will be ok even if there are a few thousand line items in the report.